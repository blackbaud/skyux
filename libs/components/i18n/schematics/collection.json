{
  "$schema": "../../../../node_modules/@angular-devkit/schematics/collection-schema.json",
  "schematics": {
    "ng-add": {
      "description": "Add @skyux/i18n to your project",
      "factory": "./ng-add/ng-add.schematic"
    },
    "lib-resources-module": {
      "description": "Generate library resources module",
      "factory": "./ng-generate/lib-resources-module/lib-resources-module.schematic",
      "schema": "./ng-generate/lib-resources-module/schema.json"
    }
  }
}
