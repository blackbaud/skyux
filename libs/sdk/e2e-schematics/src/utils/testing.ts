import {
  E2eTestRunner,
  UnitTestRunner,
  applicationGenerator,
  libraryGenerator,
} from '@nx/angular/generators';
import { Tree } from '@nx/devkit';
import { Linter } from '@nx/eslint';

export async function createTestApplication(
  tree: Tree,
  options: { name: string; e2eTestRunner?: boolean; unitTestRunner?: boolean },
): Promise<void> {
  await applicationGenerator(tree, {
    name: options.name,
    skipPackageJson: true,
    directory: `apps/${options.name}`,
    e2eTestRunner: options.e2eTestRunner
      ? E2eTestRunner.Cypress
      : E2eTestRunner.None,
    linter: Linter.None,
    unitTestRunner: options.unitTestRunner
      ? UnitTestRunner.Jest
      : UnitTestRunner.None,
  });
}

export async function createTestLibrary(
  tree: Tree,
  options: { name: string },
): Promise<void> {
  await libraryGenerator(tree, {
    name: options.name,
    skipPackageJson: true,
    directory: `libs/${options.name}`,
    linter: Linter.None,
    unitTestRunner: UnitTestRunner.None,
  });
}
