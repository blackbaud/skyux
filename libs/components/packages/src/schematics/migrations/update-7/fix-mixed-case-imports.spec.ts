import { Tree } from '@angular-devkit/schematics';
import { SchematicTestRunner } from '@angular-devkit/schematics/testing';

describe('fix mixed case imports', () => {
  let tree: Tree;
  const runner = new SchematicTestRunner(
    'schematics',
    require.resolve('../migration-collection.json')
  );
  const angularJson = {
    version: 1,
    projects: {
      test: {
        projectType: 'application',
        root: '',
        architect: {},
      },
    },
  };

  function setupTest() {
    tree = Tree.empty();
    tree.create('/angular.json', JSON.stringify(angularJson));
  }

  it('should not change import paths that are correct', async () => {
    setupTest();
    tree.create('/index.ts', `import { A } from './a';`);
    tree.create('/a.ts', `export class A {}`);
    tree.create('/empty.ts', ``);
    await runner
      .runSchematicAsync('fix-mixed-case-imports', {}, tree)
      .toPromise();
    expect(tree.readText('/index.ts')).toBe(`import { A } from './a';`);
  });

  it('should update import paths with backslashes', async () => {
    setupTest();
    tree.create('/windows.ts', "import { A } from '.\\a';");
    tree.create('/a.ts', `export class A {}`);
    await runner
      .runSchematicAsync('fix-mixed-case-imports', {}, tree)
      .toPromise();
    expect(tree.readText('/windows.ts')).toBe(`import { A } from './a';\n`);
  });

  it('should change import paths that are incorrect', async () => {
    setupTest();
    tree.create(
      '/index.ts',
      [
        `import { A } from './A';`,
        `import { C } from './path/c';`,
        `import { D } from './Path/D';`,
        `import { NonExistent } from './non';`,
        `export class B extends A {`,
        `}`,
        '',
      ].join(`\n`)
    );
    tree.create('/a.ts', `export class A {}`);
    tree.create('/Path/C.ts', `export class C {}`);
    tree.create('/Path/d.ts', `export class D {}`);
    await runner
      .runSchematicAsync('fix-mixed-case-imports', {}, tree)
      .toPromise();
    expect(tree.readText('/index.ts')).toMatchSnapshot();
  });
});
