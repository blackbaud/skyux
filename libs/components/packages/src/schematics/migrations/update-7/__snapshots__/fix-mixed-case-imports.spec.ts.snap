// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fix mixed case imports should change import paths that are incorrect 1`] = `
"import { A } from './a';
import { C } from './Path/C';
import { D } from './Path/d';
import { NonExistent } from './non';
export class B extends A {
}
"
`;
